app/src/main/kotlin/com/aliucord/manager/installer/util/ManifestPatcher.kt

```diff
@@ -25,19 +28,38 @@ object ManifestPatcher {
 
         reader.accept(object : AxmlVisitor(writer) {
             override fun child(ns: String?, name: String?) =
-                object : ReplaceAttrsVisitor(super.child(ns, name), mapOf("package" to packageName)) {
-                    private var addManagePerm = true
-
-                    override fun attr(ns: String?, name: String, resourceId: Int, type: Int, value: Any?) {
-                        super.attr(ns, name, resourceId, type, value)
-
-                        if (value == MANAGE_EXTERNAL_STORAGE) addManagePerm = false
-                    }
+                object : ReplaceAttrsVisitor(
+                    super.child(ns, name),
+                    mapOf(
+                        PACKAGE to packageName,
+                        COMPILE_SDK_VERSION to 23,
+                        COMPILE_SDK_VERSION_CODENAME to "6.0-2438415"
+                    )
+                ) {
+                    private var addExternalStoragePerm = false
 
                     override fun child(ns: String?, name: String): NodeVisitor {
                         val nv = super.child(ns, name)
 
+                        // Add MANAGE_EXTERNAL_STORAGE when necessary
+                        if (addExternalStoragePerm) {
+                            super.child(null, "uses-permission").attr(
+                                ANDROID_NAMESPACE, "name", 16842755, TYPE_STRING, Manifest.permission.MANAGE_EXTERNAL_STORAGE
```

Comment:
what's this magic number? I would store it in a const / add a comment to explain what exactly it stands for

Reply:
I have no clue what it is I just `println`ed the other uses-permission blocks and thats the resource ID it used